### ISSUE TITLE
Fatal fault in thread in Echo-Server when NET_TCP_LOG_LEVEL_DBG is enabled

### ISSUE THREAD

**Issue by bluespring** [zephyrproject-rtos/zephyr#10698]
### Overview
While running **Echo-Server** on **QEMU**, if `NET_TCP_LOG_LEVEL_DBG` is enabled, it throws an fatal error.

> Current thread ID = 0x00403cd8
> eax: 0x00007e00, ebx: 0x000006c4, ecx: 0xba0f750a, edx: 0x004050c0
> esi: 0x00411dd0, edi: 0x00000007, ebp: 0x00411da0, esp: 0x00411d90
> eflags: 0x00000202 cs: 0x0008
> call trace:
> eip: 0x000036a7
>     0x00002fea (0x270f3)
>     0x00019429 (0x270f3)
>     0x0001a4ed (0x4045b0)
>     0x0000c5e6 (0x4036c0)
>     0x0000795c (0x725e)
>     0x000018fb (0x4001a0)
>     0x0000bbec (0x4001a0)
>     0x0001b7b0 (0x4042d0)
> Fatal fault in thread 0x00403cd8! Aborting.

However, if NET_TCP_LOG_LEVEL is below DBG, there is no issue.

### Steps to reporduce
1. Enable `NET_TCP_LOG_LEVEL_DBG`
2. Run Echo-Server as instructed in [Networking with QEMU/Basic Steps](https://docs.zephyrproject.org/latest/subsystems/networking/qemu_setup.html#basic-setup)
3. Use `echo-client` in [net-tools](https://github.com/zephyrproject-rtos/net-tools) to send TCP packet
```bash
$ sudo ./echo-client -i tap0 -t 192.0.2.1
```

### Full logs
[echo-server-crash-tcp-dbg.log](https://github.com/zephyrproject-rtos/zephyr/files/2492940/echo-server-crash-tcp-dbg.log)

**Comment by bluespring** [comment]
FYI, after commenting this line, no issue is seen.
Possibly caused by `log_strdup()`.

```c
@@ -1827,32 +1830,34 @@ int net_tcp_unref(struct net_context *context)
 
 #define net_tcp_print_recv_info(str, pkt, port)                        \
        if (IS_ENABLED(CONFIG_NET_TCP_LOG_LEVEL_DBG)) {                 \
+               ARG_UNUSED(pkt);                                        \
                if (net_context_get_family(context) == AF_INET6) {      \
-                       NET_DBG("%s received from %s port %d", str,     \
+/*                     NET_DBG("%s received from %s port %d", str,     \
                                log_strdup(net_sprint_ipv6_addr(        \
                                             &NET_IPV6_HDR(pkt)->src)), \
                                ntohs(port));                           \
-               } else if (net_context_get_family(context) == AF_INET) {\
-                       NET_DBG("%s received from %s port %d", str,     \
-                               log_strdup(net_sprint_ipv4_addr(        \
+*/             } else if (net_context_get_family(context) == AF_INET) {\
+                       NET_DBG("%s received from port %d", str,        \
+                               /*log_strdup(net_sprint_ipv4_addr(      \
                                             &NET_IPV4_HDR(pkt)->src)), \
-                               ntohs(port));                           \
+                               */ntohs(port));                         \
                }                                                       \
        }
 
 #define net_tcp_print_send_info(str, pkt, port)                        \
        if (IS_ENABLED(CONFIG_NET_TCP_LOG_LEVEL_DBG)) {                 \
+               ARG_UNUSED(pkt);                                        \
                struct net_context *ctx = net_pkt_context(pkt);         \
                if (net_context_get_family(ctx) == AF_INET6) {          \
-                       NET_DBG("%s sent to %s port %d", str,           \
+/*                     NET_DBG("%s sent to %s port %d", str,           \
                                log_strdup(net_sprint_ipv6_addr(        \
                                             &NET_IPV6_HDR(pkt)->dst)), \
                                ntohs(port));                           \
-               } else if (net_context_get_family(ctx) == AF_INET) {    \
-                       NET_DBG("%s sent to %s port %d", str,           \
-                               log_strdup(net_sprint_ipv4_addr(        \
+*/             } else if (net_context_get_family(ctx) == AF_INET) {    \
+                       NET_DBG("%s sent to port %d", str,              \
+                               /*log_strdup(net_sprint_ipv4_addr(      \
                                             &NET_IPV4_HDR(pkt)->dst)), \
-                               ntohs(port));                           \
+                               */ntohs(port));                         \
                }                                                       \
        }

```

**Comment by jukkar** [comment]
Seems to be issue in logging:

This is log from echo-server:
```
[00:00:21.056,720] <inf> net_echo_server: TCP IPv6 received 1 bytes
[00:00:21.059,848] <dbg> net_tcp: (0x00403cd8): net_tcp_queue_data: [0x004045b0] Queue 0x004087d0 len 1
[00:00:21.062,645] <dbg> net_tcp: (0x00403cd8): net_tcp_trace: [0x004045b0] pkt 0x004087d0 src 4242 dst 51755
[00:00:21.062,771] <dbg> net_tcp: (0x00403cd8): net_tcp_trace:   seq 0x37a6fb39 (933690169) ack 0x3040159f (809506207/0)
[00:00:21.062,945] <dbg> net_tcp: (0x00403cd8): net_tcp_trace:   flags uAPrsf
[00:00:21.063,040] <dbg> net_tcp: (0x00403cd8): net_tcp_trace:   win 1280 chk 0xbad9
Current thread ID = 0x00403cd8
eax: 0x00007e00, ebx: 0x000006c4, ecx: 0x0f2bf983, edx: 0x004050c0
esi: 0x00411db0, edi: 0x00000007, ebp: 0x00411d80, esp: 0x00411d70
eflags: 0x00000202 cs: 0x0008
call trace:
eip: 0x000036a7
     0x00002fea (0x27113)
     0x0001944f (0x27113)
     0x0001a513 (0x4045b0)
     0x0000c60c (0x4036c0)
     0x00007982 (0x7284)
     0x000018fb (0x4001a0)
     0x0000bc12 (0x4001a0)
     0x0001b7d6 (0x4042d0)
Fatal fault in thread 0x00403cd8! Aborting.
[00:00:21.916,787] <dbg> net_tcp: (0x00404e18): tcp_retry_expired: retry 1: [0x004045b0] sent pkt 0x004087d0
```

I had gdb attached to echo-server in order to see the backtrace of the crash:
```
(gdb) file zephyr/zephyr.elf
Reading symbols from zephyr/zephyr.elf...done.
(gdb) target remote localhost:1234
Remote debugging using localhost:1234
0x0000fff0 in adjust_write_offset (timeout=<optimized out>, pos=<optimized out>, offset=0, 
    frag=<optimized out>, pkt=<optimized out>)
    at zephyr/subsys/net/ip/net_pkt.c:1465
1465				if (net_buf_tailroom(frag)) {
(gdb) c
Continuing.
^C
Program received signal SIGINT, Interrupt.
k_cpu_idle () at zephyr/arch/x86/core/cpuhalt.c:56
56	}
(gdb) info symbol 0x000036a7
log_msg_create_n + 132 in section text
(gdb) info symbol 0x00002fea
log_n + 21 in section text
(gdb) info symbol 0x0001944f
net_tcp_queue_pkt.isra + 214 in section text
(gdb) list *0x000036a7
0x36a7 is in log_msg_create_n (zephyr/include/logging/log_msg.h:384).
379				return NULL;
380			}
381	
382			msg->hdr.params.generic.ext = 1;
383			msg->payload.ext.next = cont;
384			cont->next = NULL;
385		}
386	
387		return msg;
388	}
(gdb) 
(gdb) k
Kill the program being debugged? (y or n) y
(gdb) quit
```

**Comment by jukkar** [comment]
I wonder if this is a stack space issue as the crash happens in weird place.

**Comment by jfischer-phytec-iot** [comment]
please try out if #10616 fix it

**Comment by bluespring** [comment]
@jfischer-phytec-iot I confirmed that I no longer see the crash on the stack with #10616. 
Great work!

**Comment by nashif** [comment]
fixed with #10616
